09-Time-Series-Data.ipynb

```diff
@@ -0,0 +1,786 @@
+    "\n",
+    "<pre>\n",
+    "temps = {'Brussles': 9, 'London': 3, 'Barcelona': 13, 'Rome': 16}\n",
+    "temps['Rome']\n",
+    "16\n",
+    "</pre>\n",
+    "\n",
+    "We can also find out what keywords are associated with a given dictionary, In this case:\n",
+    "\n",
+    "<pre>\n",
+    "temps.keys()\n",
+    "dict_keys(['London', 'Barcelona', 'Rome', 'Brussles'])\n",
+    "</pre>\n",
+    "\n",
+    "</code>\n",
+    "<br/>\n",
+    "</section>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can use these keys to plot specific parameters from the lightcurve, Aluminium is 'CHANNEL_3' and Zirconium is 'CHANNEL_4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false,
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "plt.figure(1, figsize=(10,5))\n",
+    "plt.plot(lyra_lightcurve.data.index, lyra_lightcurve.data['CHANNEL3'], color='blue', label='Al filter')\n",
+    "plt.plot(lyra_lightcurve.data.index, lyra_lightcurve.data['CHANNEL4'], color='red', label='Zr filter')\n",
+    "plt.ylabel('Flux (Wm^2)')\n",
+    "plt.show()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Analysing Lightcurve data\n",
+    "\n",
+    "We can asses the degree to which the lyra curves correlate with each other:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "cross_correlation = np.correlate(lyra_lightcurve.data['CHANNEL3'],\n",
+    "                                 lyra_lightcurve.data['CHANNEL4'])\n",
+    "print(cross_correlation)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Pandas\n",
+    "\n",
+    "In its own words Pandas is a Python package providing fast, flexible, and expressive data structures designed to make working with “relational” or “labeled” data both easy and intuitive. Pandas has two forms of structures, 1D series and 2D dataframe. It also has its own fucnctions associated with it.\n",
+    "\n",
+    "It is also amazing.\n",
+    "\n",
+    "Lightcurve uses these in built Pandas functions, so we can find out things like the maximum of curves:\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "max_t_goes_long = goes_lightcurve.data['xrsb'].idxmax()\n",
+    "max_t_goes_short = goes_lightcurve.data['xrsa'].idxmax()\n",
+    "max_t_lyra_al = lyra_lightcurve.data['CHANNEL3'].idxmax()\n",
+    "max_t_lyra_zr = lyra_lightcurve.data['CHANNEL4'].idxmax()\n",
+    "print('GOES long: ', max_t_goes_long)\n",
+    "print('GOES short: ', max_t_goes_short)\n",
+    "print('LYRA Al: ', max_t_lyra_al)\n",
+    "print('LYRA Zr: ', max_t_lyra_zr)"
+   ]
```

Comment:
When you type this block up in the lesson, break it up a bit with some line breaks so people can follow it more easily.